test/unit/L2/delegatorPool.test.ts

```diff
@@ -0,0 +1,246 @@
+import {FakeContract, smock} from '@defi-wonderland/smock';
+import {SignerWithAddress} from '@nomiclabs/hardhat-ethers/dist/src/signers';
+import {expect, use} from 'chai';
+
+import {ethers} from 'hardhat';
+import {DelegatorPool, DelegatorPool__factory} from '../../../typechain';
+
+use(smock.matchers);
+
+describe('DelegatorPool', function() {
+  let delegatorPool: DelegatorPool;
+
+  let bondingManagerMock: FakeContract;
+
+  let delegator: SignerWithAddress;
+  let delegator1: SignerWithAddress;
+  let delegator2: SignerWithAddress;
+  let mockL2MigratorEOA: SignerWithAddress;
+  let mockBondingManagerEOA: SignerWithAddress;
+
+  class StakeAndFees {
+    public claimedStake = 0;
+    public pendingStake;
+    public pendingFees;
+
+    public sequence = 0;
+
+    constructor(
+      public readonly initialStake: number,
+      public readonly initialFees: number,
+    ) {
+      this.pendingStake = initialStake;
+      this.pendingFees = initialFees;
+    }
+
+    calculateClaim(stake: number) {
+      const owedStake =
+        (this.pendingStake * stake) / (this.initialStake - this.claimedStake);
+
+      const owedFees =
+        (this.pendingFees * stake) / (this.initialStake - this.claimedStake);
+
+      return {
+        owedStake,
+        owedFees,
+      };
+    }
+
+    addRewards(stake: number, fees: number) {
+      this.pendingStake += stake;
+      this.pendingFees += fees;
+    }
+
+    createTx(caller: SignerWithAddress, stake: number) {
+      const {owedStake, owedFees} = this.calculateClaim(stake);
+
+      bondingManagerMock.pendingStake.returns(this.pendingStake);
+      bondingManagerMock.pendingFees.returns(this.pendingFees);
+
+      const tx = delegatorPool
+          .connect(mockL2MigratorEOA)
+          .claim(caller.address, stake);
+
+      this.claimedStake += stake;
+
+      this.pendingStake -= owedStake;
+      this.pendingFees -= owedFees;
+
+      return {tx, seq: this.sequence++, owedStake, owedFees};
+    }
+
+    async testClaim(caller: SignerWithAddress, stake: number) {
+      const {tx, seq, owedStake, owedFees} = this.createTx(caller, stake);
+      const txRec = await tx;
```

Comment:
Could we just return the value of the Promise on L60 i.e. via `const tx = await ...` instead of having an `await` here? I suspect that the event emit test on L90 will still work fine, but worth double checking.

Reply:
fixed in 64be1f3